### PR TITLE
RHEL 9.6+ Azure: enable the dataloss warning service and add unmanaged devices to the NetworkManager config

### PR DESCRIPTION
**Pull request by achilleas-k**
1. Enable the dataloss warning service
2. Add org.osbuild.nm.conf stage
3. Add NetworkManager config to set unmanaged devices

Part of #1396
These came up after a review of the currently merged changes from QE.

### CODE REVIEW

File: pkg/osbuild/nm_conf_stage.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+		settings.Keyfile == nil {
+		return fmt.Errorf("%s: at least one setting must be set", nmConfStageType)
+	}
+
+	if main := settings.Main; main != nil {
+		if err := validateArrayHasItems(main.NoAutoDefault); err != nil {
+			return fmt.Errorf("%s: main.no-auto-default %s", nmConfStageType, err)
+		}
+		if err := validateArrayHasItems(main.Plugins); err != nil {
+			return fmt.Errorf("%s: main.no-auto-default %s", nmConfStageType, err)
```
Comment (ondrejbudai) [inline pkg/osbuild/nm_conf_stage.go:72]: Copilot found out that this error message should mention `main.plugins`. It also suggested a test for it, as the `empty-main` test case only covers the first condition.

**Review by ondrejbudai — COMMENTED** [review]
A minor nitpick, otherwise lgtm. :)

File: pkg/osbuild/nm_conf_stage.go (hunk 1 above)
Comment (achilleas-k) [reply pkg/osbuild/nm_conf_stage.go]: Thanks!  Fixed.

**Review by thozza — APPROVED** [review]

**Review by ondrejbudai — APPROVED** [review]
Thanks!
